src/webAgentAPI/main.ts

```diff
@@ -1,15 +1,18 @@
 import { createAPI, setListener } from "./api";
-import { sendMessage, setInstanceId } from "./sendMessage";
+import { FDC3LocalInstance } from "./sendMessage";
 import { RegisterInstanceReturn } from "@/common/types";
 import { DesktopAgent } from "@finos/fdc3";
 
 export const createWebAgentAPI = async (): Promise<DesktopAgent> => {
-  setListener();
-
-  const registrationResult = await sendMessage("registerInstance");
+  
+  //instantiate a local instance
+  const connection = new FDC3LocalInstance();
```

Comment:
this is a good point.  I added an async 'initConnection' method to FDC3LocalInstance - so this can just be called after construction and multiple steps aren't required.